### PR TITLE
[refactor] Introduce action radio to broadcast

### PR DESCRIPTION
**Pull request by envestcc**
# Description

The `ActionRadio` component is introduced in this update to broadcast actions received in the API node to the network. 

This is a pre-PR aimed at facilitating the implement of action retry mechanism next

## Type of change
Please delete options that are not relevant.
- [] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [x] Code refactor or improvement
- [] Breaking change (fix or feature that would cause a new or changed behavior of existing functionality)
- [] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [x] make test
- [] fullsync
- [] Other test (please specify)

**Test Configuration**:
- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [] My code follows the style guidelines of this project
- [] I have performed a self-review of my code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings
- [] I have added tests that prove my fix is effective or that my feature works
- [] New and existing unit tests pass locally with my changes
- [] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: api/action_radio.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+func (ar *ActionRadio) Start() error {
+	if ar.messageBatcher != nil {
+		return ar.messageBatcher.Start()
+	}
+	return nil
+}
+
+// Stop stops the action radio
+func (ar *ActionRadio) Stop() error {
+	if ar.messageBatcher == nil {
```
Comment (CoderZhi) [inline api/action_radio.go:57]: !=

File: api/action_radio.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+			ChainID: ar.chainID,
+			Target:  nil,
+			Data:    out,
+		})
+	} else {
+		err = ar.broadcastHandler(context.Background(), ar.chainID, out)
+	}
+	if err != nil {
+		log.L().Warn("Failed to broadcast SendAction request.", zap.Error(err), zap.String("actionHash", hex.EncodeToString(hash[:])))
+	}
```
Comment (CoderZhi) [inline api/action_radio.go:89]: return err

Comment (CoderZhi) [reply api/action_radio.go:89]: nvm, it is an interface implementation

**Review by CoderZhi — APPROVED** [review]

File: api/action_radio_test.go [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+	r := require.New(t)
+	broadcastCount := uint64(0)
+	radio := NewActionRadio(
+		func(_ context.Context, _ uint32, _ proto.Message) error {
+			atomic.AddUint64(&broadcastCount, 1)
+			return nil
+		},
+		0)
+	r.NoError(radio.Start())
+	defer radio.Stop()
```
Comment (dustinxie) [inline api/action_radio_test.go:26]:
```
defer func() {
    r.NoError(radio.Stop())
}
```

**Review by dustinxie — APPROVED** [review]

**Comment by asim1019** [comment]
> # Description
> The `ActionRadio` component is introduced in this update to broadcast actions received in the API node to the network.
> 
> This is a pre-PR aimed at facilitating the implement of action retry mechanism next
> 
> ## Type of change
> Please delete options that are not relevant.
> 
> * [] Bug fix (non-breaking change which fixes an issue)
> * [] New feature (non-breaking change which adds functionality)
> * [x]  Code refactor or improvement
> * [] Breaking change (fix or feature that would cause a new or changed behavior of existing functionality)
> * [] This change requires a documentation update
> 
> # How Has This Been Tested?
> Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
> 
> * [x]  make test
> * [] fullsync
> * [] Other test (please specify)
> 
> **Test Configuration**:
> 
> * Firmware version:
> * Hardware:
> * Toolchain:
> * SDK:
> 
> # Checklist:
> * [] My code follows the style guidelines of this project
> * [] I have performed a self-review of my code
> * [] I have commented my code, particularly in hard-to-understand areas
> * [] I have made corresponding changes to the documentation
> * [] My changes generate no new warnings
> * [] I have added tests that prove my fix is effective or that my feature works
> * [] New and existing unit tests pass locally with my changes
> * [] Any dependent changes have been merged and published in downstream modules

**Pull request merged**